bmeg_app/views/literature_support_view.py

```diff
@@ -48,90 +44,134 @@
-  
+            dbc.Col(dcc.Loading(id='occr', children=html.Div()),width=3,style={"c":"100%"}),
+            dbc.Col(dcc.Loading(id='resp_histo', children=html.Div()),width=4,style={"height":"100%"}),
+            dbc.Col(dcc.Loading(id='pie_taxon', children=html.Div()),width=5,style={"height":"100%"}),
+
         ],
     ),
     html.Hr(),
-    dcc.Loading(id='evd', type="default",children=html.Div()),
+    dcc.Loading(id='evd', children=html.Div()),
 ],style={'fontFamily': styles['t']['type_font']})
 
 
 @app.callback(
-    Output("evd", "children"),
+    dash.dependencies.Output('gene_dd', 'options'),
+    [dash.dependencies.Input('gene_dd', 'search_value')]
+)
+def update_options(search_value):
+    """Lookup the search value in elastic."""
+    if not search_value:
+        raise PreventUpdate
+    genes = gene_search(search_value)
+    return genes
+
+@app.callback(
+    [Output("evd", "children"), Output("resp_histo", "children"),Output("pie_taxon", "children"), Output("occr", "children")],
     [Input('gene_dd', 'value')]
 )
-def render_callback(selected_gene):
+def build_evidence_table(search_value):
+    gene = search_value.split("/")[1]
     '''Occurance table'''
-    # Filter for gene
-    df = base_df[base_df['geneID']==selected_gene].reset_index(drop=True)
-    # Generate table
-    dictionary = lsu.parse_src_doc(df, 'litETC',['cancerType','drugAbstracts'])
-    df = lsu.build_publication_table(dictionary)
-    dashtable = dash_table.DataTable(
-        id='export_pub_table',
-        data = df.to_dict('records'),
-        columns=[{"name": i, "id": i} for i in df.columns],
+    ## associations
+    mapping = {
+        "id" : "$._gid",
+        "evidence_label" : "evidence_label",
+        "oncogenic": "oncogenic",
+        "response_type" : "response_type",
+        "source" : "source"
+    }
+    g2p = []
+    response = []
+    for row in G.query().V(gene).out("g2p_associations").render(mapping):
+        g2p.append(row.to_dict())
+        response.append(row['response_type'])
+    g2p_chem = {}
+    for row in G.query().V(gene).out("g2p_associations").as_("a").out("compounds").as_("c").render(["$a._gid", "$c._gid", "$c.synonym"]).execute():
+        if row[0] in g2p_chem:
+            g2p_chem[row[0]].append(row[2])
+        else:
+            g2p_chem[row[0]] = [row[2]]
+    g2p_cite = {}
+    for row in G.query().V(gene).out("g2p_associations").as_("a").out("publications").as_("c").render(["$a._gid", "$c.url", "$c.title"]).execute():
+        if row[0] in g2p_cite:
+            g2p_cite[row[0]].append( (row[1], row[2]) )
+        else:
+            g2p_cite[row[0]] = [(row[1], row[2])]
+
+
+    rows = []
+    for g in g2p:
+        if g["id"] in g2p_cite:
+            for c in g2p_cite[g["id"]]:
+                row = dict(**g)
+                if row["id"] in g2p_chem:
+                    row['compound'] = ",".join(list(i for i in g2p_chem[row["id"]] if i))
```

Comment:
cleaner compound names (capitalize drug name and add space between joins)

`row['compound'] = ", ".join(list(i.capitalize() for i in g2p_chem[row["id"]] if i))`